integration-tests/utils/lyle-utils.js

```diff
@@ -19,13 +19,26 @@ const fetchLyle = async (path, email) => {
   const url = `${lyleUrl}/api/${path}`
   const authClient = await makeAuthClient()
 
-  const res = await fetch(url, {
+  return await fetch(url, {
     method: 'POST',
     headers: { 'Content-Type': 'application/json', ...(await authClient.getRequestHeaders(url)) },
     body: JSON.stringify({ email })
   })
-
-  return res.json()
+    .then(async resp => {
+      if (resp.ok) {
+        // response.status >= 200 && response.status < 300
+        return resp.json()
+      }
+      const respText = await resp.text()
+      console.error(`fetch Lyle response: ${respText}`)
+      return respText
+    })
+    .catch(async err => {
+      console.error(err)
+      const errorBody = await err.response.text()
+      console.error(`fetch Lyle request response body: ${errorBody}`)
+      throw err
+    })
```

Comment:
This can also be written with try/catch which is more consistent with this codebase and is generally easier to read and reason about.
```
  try {
    const res = await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json', ...(await authClient.getRequestHeaders(url)) },
      body: JSON.stringify({ email })
    })
    if (res.ok) {
      // response.status >= 200 && response.status < 300
      return res.json()
    } else {
      // delegate non-2xx response to enclosing try/catch
      throw { response: res, status: res.status }
    }
  } catch (err) {
    console.error(err)
    const errorBody = await err.response.text()
    console.error(`fetch Lyle request response body: ${errorBody}`)
    throw err
  }
```
Also, delegating to the outer try/catch on non-2xx responses eliminates some code duplication.

Reply:
This is nice. Code consistency is important. I'll update to use `try/catch`. 
I have one concern. I don't think we should `throw { response: res, status: res.status }` in `else` block. I prefer just to log response text and returns it. Then it can be used for negative testing where non-json response body is expected in fetch. Albeit I don't see any usage like that now. :)
What do you think if I keep the `else` code unchanged?

Reply:
The expected response from this function is JSON, so I'm against returning plain text. Also, this function is purely in support of these tests and I don't see a lot of value in performing negative tests of this function in isolation.